### PR TITLE
Introduce B2B Resolving Capabilities to Email/SMS Template Mgt APIs

### PR DESCRIPTION
**Pull request by dhaura**
### Purpose
- This PR modifies the following APIs and their endpoints with `resolve` query parameter, so that when `resolve` param is set to true, the endpoint will return the resolved notification template throughout its ancestor organization hierarchy. 
    - Email Templates V2 API
        - `GET` /email/template-types/{template-type-id}/app-templates/{app-uuid}
        - `GET` /email/template-types/{template-type-id}/app-templates/{app-uuid}/{locale}  
    - Notification Templates API
        - `GET` /notification/email/template-types/{template-type-id}/org-templates
        - `GET` /notification/email/template-types/{template-type-id}/org-templates/{locale}
        - `GET` /notification/email/template-types/{template-type-id}/app-templates/{app-uuid}
        - `GET` /notification/email/template-types/{template-type-id}/app-templates/{app-uuid}/{locale}
        - `GET` /notification/sms/template-types/{template-type-id}/org-templates
        - `GET` /notification/sms/template-types/{template-type-id}/org-templates/{locale}
        - `GET` /notification/sms/template-types/{template-type-id}/app-templates/{app-uuid}
        - `GET` /notification/sms/template-types/{template-type-id}/app-templates/{app-uuid}/{locale}
- If the `resolve` param is set to false, or not set at all, it will return only the current organization's or application's template. Threrefore, if the template is not found it will return `404`.

### Related Issues
- https://github.com/wso2/product-is/issues/21214

### Related PRs
- https://github.com/wso2-extensions/identity-governance/pull/887
- https://github.com/wso2-extensions/identity-event-handler-notification/pull/291

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -820,12 +820,12 @@
         <version.org.wso2.orbit.javax.xml.bind>2.3.1.wso2v1</version.org.wso2.orbit.javax.xml.bind>
         <maven.buildnumber.plugin.version>1.4</maven.buildnumber.plugin.version>
         <org.apache.felix.annotations.version>1.2.4</org.apache.felix.annotations.version>
-        <identity.governance.version>1.11.11</identity.governance.version>
+        <identity.governance.version>1.11.18</identity.governance.version>
```
Comment (dhaura) [inline pom.xml:823]: Bump governance version once https://github.com/wso2-extensions/identity-governance/pull/887 is merged.

File: pom.xml [hunk 2]
```diff
@@ -820,12 +820,12 @@
...
         <carbon.identity.framework.version>7.7.19</carbon.identity.framework.version>
         <maven.findbugsplugin.version>3.0.5</maven.findbugsplugin.version>
         <findsecbugs-plugin.version>1.12.0</findsecbugs-plugin.version>
         <maven.checkstyleplugin.excludes>**/gen/**/*</maven.checkstyleplugin.excludes>
-        <identity.event.handler.version>1.9.12</identity.event.handler.version>
+        <identity.event.handler.version>1.9.24</identity.event.handler.version>
```
Comment (dhaura) [inline pom.xml:828]: Bump event handler version once https://github.com/wso2-extensions/identity-event-handler-notification/pull/291 is merged.

**Comment by AnuradhaSK** [comment]
Let's link the API doc impact issue and send the doc update PR as well

**Comment by dhaura** [comment]
### Docs
- Issue - https://github.com/wso2/product-is/issues/21947

**Comment by dhaura** [comment]
### Integration Tests
- Integration tests related to this improvement are tracked with https://github.com/wso2/product-is/issues/21973

File: pom.xml (hunk 2 above)
Comment (dhaura) [reply pom.xml]: Bumped with https://github.com/wso2/identity-api-server/pull/755/commits/6d4675c3dfa2519ead38020dd79865ab1138f2b3

File: pom.xml (hunk 1 above)
Comment (dhaura) [reply pom.xml]: Bumped with https://github.com/wso2/identity-api-server/pull/755/commits/6d4675c3dfa2519ead38020dd79865ab1138f2b3

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**